maven/impl-maven-embedded/src/test/java/org/jboss/shrinkwrap/resolver/impl/maven/embedded/MavenDownloadsTestCase.java

```diff
@@ -56,6 +64,53 @@ public void testDownloadFromGivenDestinationWithoutUsingCacheAndExtract() throws
         verifyExtraction(2, "resolver-3.0.0-alpha-1", "resolver-3.0.0-alpha-3");
     }
 
+    @Test
+    public void testDownloadInMultipleThreads() throws IOException, InterruptedException {
+        // cleanup
+        FileUtils.deleteDirectory(targetMavenDir);
+        final CountDownLatch latch = new CountDownLatch(1);
+
+        // multiple download
+        createThreadWithDownload(latch).start();
+        createThreadWithDownload(latch).start();
+        latch.countDown();
+        downloadSWR300Alpha1();
```

Comment:
Shouldn't we do the actual download in the threads? Also `countDown` doesn't look to me as it's in the right place.